src/appleseed/renderer/meta/tests/test_energycompensation.cpp

```diff
@@ -0,0 +1,48 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.foundation headers.
+#include "foundation/utility/test.h"
+
+namespace renderer
+{
+
+// From renderer/modeling/bsdf/microfacethelper.cpp.
+void write_microfacet_directional_albedo_tables_to_exr(const char* directory);
```

Comment:
I suggest moving this to `microfacethelper.h`.

Reply:
I had it like that, but because this is only useful in the unit test, I thought it would be cleaner not to add it to the header and use a prototype instead.

Reply:
It's defined in `microfacethelper.cpp` but not exposed in `microfacethelper.h` so it looks odd (and could be removed by accident, only to find out at link time it was actually used). It would also be the first time we do this. Finally, there are many cases already where we expose stuff just for testing purposes, which I think is just fine. In conclusion I'd suggest moving the declaration to `microfacethelper.h` if you agree :)